File: 0169-majority-element/0169-majority-element.py
import collections
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        # 분할한 부분에서 과반을 찾아서 계속 위로 올려보내는 방식
        if not nums:
            return None
        if len(nums) == 1:
            return nums[0]
        
        half = len(nums)//2
        a = self.majorityElement(nums[:half])
        b = self.majorityElement(nums[half:])
        print([b,a])
        return [b,a][nums.count(a)>half]
        

File: 0169-majority-element/0169-majority-element.c
int majorityElement(int* nums, int numsSize){
    int major = nums[0];
    int count = 1;
    for (int i=1; i<numsSize; i++){
        if(count == 0){
            count ++;
            major = nums[i];
        } else if (major == nums[i]){
            count ++ ;
        } else {count--;}
    }
    return major;
}